### PR TITLE
Jbibla/user menu

### LINKED ISSUE #4306: Implement user account indicator
Part of #4237 

Works started in #4275

### PR DESCRIPTION
**Pull request by jbibla**
Closes #4306 

just a start

### CODE REVIEW

**Comment by faboweb** [comment]
🚀 Deployed on https://5ee39400d18e7456848e1c3d--lunieio.netlify.app

File: app/src/components/common/Address.vue [hunk 1]
```diff
@@ -21,34 +20,41 @@ import { formatAddress } from "src/filters"
...
+    copySuccess: false
   }),
   methods: {
     onCopy() {
       this.copySuccess = true
       setTimeout(() => {
         this.copySuccess = false
       }, 2500)
-    },
-  },
```
Comment (Bitcoinera) [inline app/src/components/common/Address.vue:42]: Nooo, why is this changing again?

**Comment by faboweb** [comment]
Either add "WIP" to title or make it a draft PR

File: app/src/components/common/Address.vue (hunk 1 above)
Comment (faboweb) [reply app/src/components/common/Address.vue:42]: probably dependencies not updated?

Comment (Bitcoinera) [reply app/src/components/common/Address.vue:42]: ah yeah, that must be it. Older version of prettier. Ping @jbibla

**Comment by Bitcoinera** [comment]
I am not as good as you are with styles Jordan 😆

**Comment by faboweb** [comment]
@mariopino can you help with styling?

**Comment by Bitcoinera** [comment]
image

🔥 🤩 💪

**Comment by faboweb** [comment]
Looks amazing. Everything hidden between experimental flag apart from logout?

**Comment by Bitcoinera** [comment]
> Looks amazing. Everything hidden between experimental flag apart from logout?

Everything is hidden now behind `developmentMode`. Notifications are then further hidden inside `experimentalMode`

**Comment by Bitcoinera** [comment]
TODOS (from my point of view)

- get user email to display it on the userMenu (instead of `account1`)

- fix this Avatar error: `Expected String with value "null", got Null`

- couple of fixes for addresses inside UserMenu

**Comment by Bitcoinera** [comment]
The rest is pretty much done. The whole flow with sign in and sign out through this new component

**Comment by Bitcoinera** [comment]
Going for images now!

**Comment by Bitcoinera** [comment]
XD

image

**Comment by mariopino** [comment]
> image
> 
> 🔥 🤩 💪

Oh so beautiful!!!

**Comment by faboweb** [comment]
How to test this? Doesn't appear in experimental mode

**Comment by Bitcoinera** [comment]
> How to test this? Doesn't appear in experimental mode

You cannot see it in the deploy? :eyes:  Let me check

**Comment by Bitcoinera** [comment]
Still, @mariopino said he would like to add this design: https://www.sketch.com/s/8ffba04e-5055-44dd-984c-c60a74f5103b/a/mjv4KL

**Comment by Bitcoinera** [comment]
Still failing on the ValidatorList not updating in the E2E?

**Comment by Bitcoinera** [comment]
It seems something is broken with the netlify deployement. The code there doesn't match the one in this PR, that is why it is impossible to test it on the preview. It is no real preview, no `UserMenu` component there:

image

**Comment by Bitcoinera** [comment]
Now it should be possible to test it on the preview

**Comment by Bitcoinera** [comment]
> Now it should be possible to test it on the preview

EDIT: just tried and yes, working like a charm

File: app/src/vuex/modules/session.js [hunk 2]
```diff
@@ -60,6 +60,7 @@ export default ({ apollo }) => {
     },
     setExperimentalMode(state) {
       state.experimentalMode = true
+      state.developmentMode = true
```
Comment (faboweb) [inline app/src/vuex/modules/session.js:63]: ?

Comment (faboweb) [reply app/src/vuex/modules/session.js:63]: why are doing 2 modes?

Comment (Bitcoinera) [reply app/src/vuex/modules/session.js:63]: otherwise I'd need to add a "develoment" flag so this kind of things can be tested on a preview

Comment (Bitcoinera) [reply app/src/vuex/modules/session.js:63]: But I can do that. What you prefer

Comment (faboweb) [reply app/src/vuex/modules/session.js]: I switched it to just display in experimental mode

File: app/src/components/account/UserMenu.vue [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+    selectedOption: "",
+  }),
+  computed: {
+    ...mapState([`session`, `account`]),
+    ...mapGetters([`address`, `currentNetwork`]),
+    user() {
+      return this.account.userSignedIn ? this.account.user : null
+    },
+  },
+  watch: {
```
Comment (jbibla) [inline app/src/components/account/UserMenu.vue:141]: hey @Bitcoinera why do we need a `watch` here?

Comment (Bitcoinera) [reply app/src/components/account/UserMenu.vue:141]: every time a new address is registered on connection it gets added to the UserMenu. Right now it is very primitive though and it serves only as a mean of displaying them (and get the cool pics like the last one I posted 😛 )

But the idea is to, in another PR, you would be able to jump from one to another just by clicking it.

I could also add that one here

**Comment by Bitcoinera** [comment]
Mmm, this just grew too big imo...



Another unexpected result from the present styles changes:



Will be fixing these now

**Comment by Bitcoinera** [comment]
Magic link sent modal based on [Charlie's design](https://www.sketch.com/s/8ffba04e-5055-44dd-984c-c60a74f5103b/a/mjv4KL):

**Comment by jbibla** [comment]
> Another unexpected result from the present styles changes:

@Bitcoinera did you add the user box back to the sidebar? i'm pretty sure i removed it.

File: app/src/components/account/UserMenu.vue [hunk 4]
```diff
@@ -43,12 +47,18 @@
             />
             <div>
               <span class="address">{{ address.address | formatAddress }}</span>
+              <span class="address-network">{{ address.networkId }}</span>
```
Comment (jbibla) [inline app/src/components/account/UserMenu.vue:50]: i saw your comments in slack but i don't think IDs is the right solution. maybe network titles (why IDs?!) - but maybe a little testnet tag would work too.

File: app/src/components/account/UserMenu.vue [hunk 5]
```diff
@@ -133,12 +144,12 @@ export default {
...
       this.selectedAddress = address.address
-      const selectedNetwork = this.networks.find(
+      this.selectedNetwork = this.networks.find(
         (network) => network.id === address.networkId
       )
-      this.$store.dispatch(`setNetwork`, selectedNetwork)
+      this.$store.dispatch(`setNetwork`, this.selectedNetwork)
       this.$store.dispatch(`signIn`, {
-        sessionType: `explore`,
+        sessionType: address.sessionType,
```
Comment (jbibla) [inline app/src/components/account/UserMenu.vue:152]: nice. thanks.

**Comment by jbibla** [comment]
thanks for the fixes ana. can we merge and open a new ticket for the remaining issues?

File: app/src/components/account/UserMenu.vue (hunk 4 above)
Comment (Bitcoinera) [reply app/src/components/account/UserMenu.vue:50]: We could add that cool purple test icon Charlie made. I should have it somewhere... or maybe lost it :sweat_smile:

**Comment by Bitcoinera** [comment]
> thanks for the fixes ana. can we merge and open a new ticket for the remaining issues?

I think we can do it here no? It is not so much left to do. Just add the icon to SessionFrame and the test icon for networks

**Comment by Bitcoinera** [comment]
OK, so this is how it is looking now:

UserMenu with testnet icons:

image


SingInModal with SessionFrame but beautiful icons from Charlie's designs and added by Mario:

image


MagicLinkSentModal:

image


Let's merge :woman_dancing:

**Comment by faboweb** [comment]
e2e fails:

``` ✖ signin.spec
 – Sign in with local account (23.444s)
   Timed out while waiting for element <#sign-in> to be present for 10000 milliseconds. - expected "visible" but got: "not found" (10474ms)
       at signIn (/home/runner/work/lunie/lunie/app/tests/e2e/signin.spec.js:95:17)
   SKIPPED:
   - Import local account
```

**Comment by Bitcoinera** [comment]
> e2e fails:

Yes, we also need to fix the E2E. I can do it

File: app/tests/e2e/signin.spec.js [hunk 6]
```diff
@@ -92,8 +92,8 @@ async function signOut(browser) {
 
 async function signIn(browser) {
   await openMenu(browser)
-  await browser.waitForElementVisible("#sign-in", 10000, true)
-  await browser.click("#sign-in")
+  await browser.waitForElementVisible(".trigger", 10000, true)
```
Comment (faboweb) [inline app/tests/e2e/signin.spec.js:95]: this is not readable can we give  the elements an id? same for the next line

Comment (Bitcoinera) [reply app/tests/e2e/signin.spec.js:95]: the issue here for E2E is that one thing is what we see in the code and a very different what is actually rendered due to the popover element. 

Oh, but you meant something else. Ok, I see. Let's try. It is a bit confusing now with this popover thingy

File: app/src/components/account/UserMenu.vue [hunk 7]
```diff
@@ -11,7 +11,7 @@
     </router-link>
     <v-popover open-class="user-menu-popover">
       <!-- This will be the popover target (for the events and position) -->
-      <div class="avatar-container">
+      <div id="trigger" class="avatar-container">
```
Comment (faboweb) [inline app/src/components/account/UserMenu.vue:14]:
```suggestion
      <div id="open-user-menu" class="avatar-container">
```

File: app/tests/e2e/signin.spec.js [hunk 8]
```diff
@@ -92,8 +92,8 @@ async function signOut(browser) {
 
 async function signIn(browser) {
   await openMenu(browser)
-  await browser.waitForElementVisible(".trigger", 10000, true)
-  await browser.click(`.menu-list-item span:nth-of-type(1)`)
+  await browser.waitForElementVisible("#trigger", 10000, true)
```
Comment (faboweb) [inline app/tests/e2e/signin.spec.js:94]:
```suggestion
  await browser.waitForElementVisible("#open-user-menu", 10000, true)
```

**Comment by jbibla** [comment]
lol 💪
Screen Shot 2020-07-03 at 8 58 12 AM

**Comment by Bitcoinera** [comment]
> lol 
> Screen Shot 2020-07-03 at 8 58 12 AM

XDD

**Comment by Bitcoinera** [comment]
Good to go. Let's merge this one! :muscle:

**Pull request merged**